src/TestableIO.System.IO.Abstractions.TestingHelpers/MockDirectory.cs

```diff
@@ -530,7 +530,38 @@ public override void Move(string sourceDirName, string destDirName)
         /// <inheritdoc />
         public override IFileSystemInfo ResolveLinkTarget(string linkPath, bool returnFinalTarget)
         {
-            throw CommonExceptions.NotImplemented();
+            var initialContainer = mockFileDataAccessor.GetFile(linkPath);
+            if (initialContainer.LinkTarget != null)
+            {
+                var nextLocation = initialContainer.LinkTarget;
+                var nextContainer = mockFileDataAccessor.GetFile(nextLocation);
+
+                if (returnFinalTarget)
+                {
+                    // The maximum number of symbolic links that are followed:
+                    // https://learn.microsoft.com/en-us/dotnet/api/system.io.directory.resolvelinktarget?view=net-6.0#remarks
+                    int maxResolveLinks = XFS.IsWindowsPlatform() ? 63 : 40;
+                    for (int i = 1; i < maxResolveLinks; i++)
+                    {
+                        if (nextContainer.LinkTarget == null)
+                        {
+                            break;
+                        }
+                        nextLocation = nextContainer.LinkTarget;
+                        nextContainer = mockFileDataAccessor.GetFile(nextLocation);
+                    }
```

Comment:
Would it make sense to add a test case for this?

Reply:
Yes, of course!
I added a test (and fixed the error in the implementation).